src/Driver.cpp

```diff
@@ -1,598 +1,558 @@
-    DeviceStats  device_stats;
-    device_stats.dropCount = getDropCounter();
-    device_stats.overflowCount = getOverflowCounter();
-    device_stats.localRemoteBitrate = getLocalRemoteBitrate();
-    device_stats.remoteLocalBitrate = getRemoteLocalBitrate();
-    device_stats.receivedSignalStrengthIndicator = getReceivedSignalStrengthIndicator();
-    device_stats.signalIntegrityLevel = getSignalIntegrityLevel();
-    device_stats.propagationTime = getPropagationTime();
-    device_stats.relativeVelocity = getRelativeVelocity();
-    return device_stats;
-}
-
-int Driver::getIntValue(std::string value_name){
-    std::stringstream ss;
-    ss << "+++" << value_name;
-    std::string s = ss.str();
-    sendWithLineEnding(s);
-    return waitSynchronousInt(s.substr(3, s.size()));
+: iodrivers_base::Driver(max_packet_size)
+{
+    mode = DATA;
 }
-DeliveryStatus Driver::getInstantMessageDeliveryStatus(){
-    return currentInstantMessage.deliveryStatus;
+
+Driver::~Driver()
+{
 }
 
-Position Driver::getPosition(bool x){
-    //TODO maybe make both variants possible polling and asynch messages 
-    //important a
+//bool Driver::sendData(bool &mailCommand)
+//{
+//    bool ret = sendCommand(mailCommand);
+//    if(mode == DATA)
+//    {
+//        ret |= sendRawData();
+//    }
+//    return ret;
+//}
 
-    if(reverse_mode == REVERSE_POSITION_RECEIVER){
-	std::cout << "usbl driver.cpp is position receiver!\n";
-    std::cout << "now returning X: " << current_position.x << std::endl;
-    std::cout << "now returning Y: " << current_position.y << std::endl;
-    std::cout << "now returning Z: " << current_position.z << std::endl;
+WaitResponse Driver::sendCommand(void)
+{
+    if(!queueCommand.empty())
+    {
+        std::string command = queueCommand.front();
 
-        new_position_available = false;
+        std::cout << "Write Line: " << command << std::endl;
+        std::string buffer = fillCommand(command);
 
-	return current_position;
-    }
-    else{
-	std::cout << "usbl driver.cpp is NOT A position receiver! reverse_mode = " << reverse_mode << " where sender = " << REVERSE_POSITION_SENDER << " and recv = " << REVERSE_POSITION_RECEIVER <<"\n";
-
-      std::string position_string;
-      if (x){
-        sendWithLineEnding("+++AT?UPX");
-        position_string = waitSynchronousString("AT?UPX");
-      } else {
-        sendWithLineEnding("+++AT?UP");
-        position_string = waitSynchronousString("AT?UP");
-      }
-      return UsblParser::parsePosition(position_string);
+        iodrivers_base::Driver::writePacket(reinterpret_cast<const uint8_t*>(buffer.c_str()), buffer.length());
+
+        return modeManager(command);
     }
-//    return current_position;
+    return NO_RESPONSE_REQUIRED;
 }
 
-int Driver::getSystemTime(){
-    sendWithLineEnding("+++AT?UT");
-    return waitSynchronousInt();
+bool Driver::sendRawData(void)
+{
+    if(!queueRawData.empty())
+    {   // TODO Need to check the size of the Raw data?
```

Comment:
If this could be the case, you need to move this comment at the place the data is queued, not where it is consumed.